ipfs.c

```diff
@@ -340,14 +349,19 @@ void send_add_reader_req_via_TCP(SST_ctx_t *ctx, char *add_reader) {
             unsigned char encrypted_entity_nonce[encrypted_entity_nonce_length];
             memcpy(encrypted_entity_nonce, data_buf + key_size * 2,
                    encrypted_entity_nonce_length);
-            save_distribution_key(data_buf, data_buf_length, ctx, key_size);
+            save_distribution_key(data_buf, ctx, key_size);
             unsigned int decrypted_entity_nonce_length;
-            unsigned char *decrypted_entity_nonce =
-                symmetric_decrypt_authenticate(
+            unsigned char *decrypted_entity_nonce;
```

Comment:
Ditto for NULL initialization.

Reply:
Fixed here. 104101a717b54ac251a8a2ea8b41c51660cbf320